Add more information on audit/prove benchmarking. Link forum posts (#402)
subspace/subspace-docs: docs/farming-&-staking/farming/advanced-cli/cli-tips.mdx

Replace /path/to/your/plot with the actual path to your plot. This will provide you with metrics and insights regarding plotting performance.

At the moment, Subspace Node supports two implementation mechanisms:

- `single`, that opens the file once and uses that for reading from multiple threads.
- `rayon`, that opens files as many times as there are farming threads and, for each thread, uses a dedicated file handle. **Rayon** is the library we use for multi-threading stuff.

> To interpret the results: typically, you assess throughput to determine the required size of the SSD for efficient auditing. Both CPU and disk performance influence this metric, so adding a second disk of the same size may not necessarily double the time; its impact is likely to be less significant. Nonetheless, this approach provides you with a general understanding of performance.

Reviewer: ```suggestion
To interpret the results: typically, you assess throughput to determine the theoretical limits of your environment. Both CPU and disk performance influence this metric, so adding a second disk of the same size may not necessarily double the time; its impact is likely to be less significant. Nonetheless, this approach provides you with a general understanding of performance.
```

@nazar-pc is this closer to what we get from this feature? It feels a little off still? Is "determine the theoretical maximum plot size your environment can audit" better?

Reviewer 2: It is max auditable size, most likely across any number of disks. I think we should also remove `single` benchmark from there since we do not use it (anyone care to send a PR?).

Reviewer: @nazar-pc struggling to find what you're referring to here:

> I think we should also remove `single` benchmark from there since we do not use it (anyone care to send a PR?).

My last comment above was a question but I missed the question mark.

I also notice that we only have `benchmark audit` and no `benchmark prove` mentioned in [readme.md](https://github.com/subspace/subspace/blob/912aa3ba2bb3d2e94ecf0425f35179a838efa883/crates/subspace-farmer/README.md?plain=1#L74) and [farming.md](https://github.com/subspace/subspace/blob/912aa3ba2bb3d2e94ecf0425f35179a838efa883/docs/farming.md?plain=1#L322). Do you want us to add `prove` to these at the same time?

Reviewer: OK, I think [this](https://github.com/subspace/subspace/pull/2465) is what Nazar means - remove the `single` parameter.

Reviewer 2: `prove` should have been there too, yes. https://github.com/subspace/subspace/pull/2465 indeed disables `single` benchmark by default as most people don't care about it.

To read more about audit benchmarking, [read this forum post](https://forum.subspace.network/t/audit-benchmark-result-explanation-please/2149).

There is a second command available, which helps you determine how much time your farmer has after auditing to provide proof.

```
./subspace-farmer benchmark prove /path/to/your/plot
```

Each farmer has ~4 seconds to audit and prove, so depending on how fast auditing is, the remaining time will be used for proving.
Proving performance doesn’t depend on the plot size.

To read more about prove benchmarking, [read this forum post](https://forum.subspace.network/t/audit-benchmark-result-explanation-please/2150).

> 

Reviewer: Could we add a line about this log message please? We quite often see community members asking what it means.

> If an environment does not have enough time for the proving step, a message such as this is logged by the farmer: 
> ```
> Proving for solution skipped due to farming time limit slot=6723846 sector_index=46
> ```

Author: Excellent idea, added!
